core/primitives/src/runtime/config.rs

```diff
@@ -51,30 +47,54 @@ impl RuntimeConfig {
             account_creation_config: AccountCreationConfig::default(),
         }
     }
+}
+
+/// An actual runtime configuration the node will use.
+///
+/// This is constructed from the runtime configuration given in the genesis file
+/// but i) may have it’s `max_gas_burnt_view` limit adjusted and ii) provides
+/// a method which returns configuration with adjustments done through protocol
+/// version upgrades.
+pub struct ActualRuntimeConfig {
+    /// The runtime configuration taken from the genesis file but with possibly
+    /// modified `max_gas_burnt_view` limit.
+    runtime_config: Arc<RuntimeConfig>,
+
+    /// The runtime configuration with lower storage cost adjustment applied.
+    with_lower_storage_cost: Arc<RuntimeConfig>,
+}
+
+impl ActualRuntimeConfig {
+    /// Constructs a new object from specified genesis runtime config.
+    ///
+    /// If `max_gas_burnt_view` is provided, the property in wasm limit
+    /// configuration will be adjusted to given value.
+    pub fn new(genesis_runtime_config: RuntimeConfig, max_gas_burnt_view: Option<Gas>) -> Self {
+        let mut config = genesis_runtime_config;
+        if let Some(gas) = max_gas_burnt_view {
+            config.wasm_config.limit_config.max_gas_burnt_view = gas;
+        }
+        let runtime_config = Arc::new(config.clone());
+
+        // Adjust as per LowerStorageCost protocol feature.
+        config.storage_amount_per_byte = 10u128.pow(19);
+        let with_lower_storage_cost = Arc::new(config);
```

Comment:
We cannot do this and need to preserve the original semantics. Otherwise some old blocks won't be able to be applied

Reply:
The lower storage is only used in protocol version which enabled it. See `for_protocol_version` function.